solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -146,14 +158,82 @@ public SolrQueryRequest getRequest() {
   @Override
   public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, float boost)
       throws IOException {
-    final FeatureWeight[] allFeatureWeights = createWeights(meta.getAllFeatures(),
-        searcher, needsScores);
-    final FeatureWeight[] modelFeaturesWeights = createWeights(meta.getFeatures(),
-        searcher, needsScores);
-
-    return new ModelWeight(searcher, modelFeaturesWeights, allFeatureWeights);
+    boolean newCode = true;
+    if (!newCode){
+      long  start = System.currentTimeMillis();
+      final FeatureWeight[] allFeatureWeights = createWeights(meta.getAllFeatures(),
+          searcher, needsScores);
+      final FeatureWeight[] modelFeaturesWeights = createWeights(meta.getFeatures(),
+          searcher, needsScores);
+      long  end = System.currentTimeMillis();
+      log.info("In ModelQuery.java::createWeight() OLD createWeight time:{} ms", (end-start));
+      return new ModelWeight(searcher, modelFeaturesWeights, allFeatureWeights);
+    }
+    else{      
+      long  start = System.currentTimeMillis();
```

Comment:
ultimately I don't think we'll want to perf log production, although perhaps we can discuss this?